metaflow/plugins/aws/eks/kubernetes.py

```diff
@@ -337,8 +337,32 @@ def _print_available(tail, stream, should_persist=False):
         _print_available(stdout_tail, "stdout")
         _print_available(stderr_tail, "stderr")
 
-        if self._job.has_failed:
-            exit_code, reason = self._job.reason
+        try:
+            # Now the pod is no longer running, but we need to wait for the job
+            # status to update. *Usually* that happens pretty much immediately,
+            # but since it is technically done by the job controller
+            # asynchronously, there's room for a race condition where pod is
+            # done but the job is still "active". It is more likely to happen
+            # when the control plane is overloaded, e.g. on minikube/kind.
+            self._job.wait_done(timeout_seconds=20)
+        except TimeoutError:
```

Comment:
Is it feasible to encapsulate this logic in `kubernetes_client.py`? Currently, that module takes care of all the complexity of ascertaining the correct job status.

Reply:
Let me see, I now get your intent that `RunningJob` was supposed to abstract away k8s job and pod together